src/components/ComparisonMap/ComparisonMap.vue

```diff
@@ -34,34 +80,70 @@
         type: Array,
         required: true,
       },
+      timeSeries: {
+        type: Array,
+        default: () => [],
+      },
     },
 
     data () {
       return {
-        oldSatelliteImageUrl: '',
-        satelliteImageUrl: '',
+        dateIndex: -1,
+        oldDateIndex: -1,
+        isLoadingSatelliteImages: false,
       }
     },
 
     computed: {
       date () {
-        return new Date()
+        return this.timeSeriesDates[this.dateIndex]
       },
       oldDate () {
-        const date = new Date()
-        date.setFullYear(date.getFullYear() - 1)
-        return date
+        return this.timeSeriesDates[this.oldDateIndex]
+      },
+      timeSeriesDates () {
+        return this.timeSeries.map(item => new Date(item[0]))
+      },
+      previousDateIndex () {
+        return this.dateIndex - 1
+      },
+      nextDateIndex () {
+        return this.dateIndex + 1
+      },
+      previousDate () {
+        return this.timeSeriesDates[this.previousDateIndex]
+      },
+      nextDate () {
+        return this.timeSeriesDates[this.nextDateIndex]
+      },
+      previousOldDateIndex () {
+        return this.oldDateIndex - 1
+      },
+      nextOldDateIndex () {
```

Comment:
Old here is referring to last year? Can you describe that in a comment?

Reply:
```
// On mounted, `date` is the last date in the time series
// `oldDate` is the closest date in the time series one year before `date`
```
I will add that comment to the `mounted` method. It's hard to find a good name, because both dates will change if the user clicks on the arrows, so this is only true when we initialise it.